ianalyzer_readers/readers/core.py

```diff
@@ -68,25 +66,50 @@ def __init__(self,
         self.required = required
         self.skip = skip
 
-
-class Reader(object):
+class Reader:
     '''
     A base class for readers. Readers are objects that can generate documents
     from a source dataset.
 
-    Subclasses of `Reader` can be created to read particular data formats or even
-    particular datasets.
+    Subclasses of `Reader` can be created to read specific data formats. 
+    In practice, you will probably work with a subclass of `Reader` like `XMLReader`,
+    `CSVReader`, etc., that provides the core functionality for a file type, and create
+    a subclass for a specific dataset.
     
-    The `Reader` class is not intended to be used directly. Some methods need to
-    be implemented in child components, and will raise `NotImplementedError` if
-    you try to use `Reader` directly.
+    Some methods of this class need to be implemented in child classes, and will raise
+    `NotImplementedError` if you try to use `Reader` directly.
 
     A fully implemented `Reader` subclass will define how to read a dataset by
     describing:
 
     - How to obtain its source files.
-    - What fields each document contains.
-    - How to extract said fields from the source files.
+    - How to parse and iterate over source files.
+    - What fields each document contains, and how to extract them from the source data.
+
+    This requires implementing the following attributes/methods:
+
+    - `fields`: a list of `Field` instances that describe the fields that will appear in
+        documents, and how to extract their value.
+    - `sources`: a method that returns an iterable of sources (e.g. file paths), possibly
+        with metadata for each.
+    - `data_directory` (optional): a string with the path to the directory containing
+        the source data. You can use this in the implementation of `sources`; it's not
+        used elsewhere.
```

Comment:
Isn't `data_directory`, strictly speaking, mandatory? Without it being defined, `load_corpora` will not mark the corpus as ready for indexing. Or has this been changed?

Reply:
I-analyzer uses a subclass of `Reader` (`CorpusDefinition`) which implements `data_directory`. (Not sure if it's mandatory? But it is validated/specified, at least.) But it's never actually used in this package. To be honest, I'm not sure if it should be documented at all here.